react/features/google-api/googleApi.native.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import {
+    API_URL_BROADCAST_STREAMS,
+    API_URL_LIVE_BROADCASTS
+} from './constants';
+
+/**
+ * Class to encapsulate Google API functionalities and provide a similar
+ * interface to what WEB has. The methods are different, but the point is that
+ * the export object is similar so no need for different export logic.
+ */
+class GoogleApi {
+
+    /**
+     * Retrieves the available YouTube streams the user can use for live
+     * streaming.
+     *
+     * @param {Object} googleAuth - The response from a successful Google Auth.
+     * @returns {Promise}
+     */
+    getYouTubeLiveStreams(googleAuth: Object): Promise<*> {
+        return new Promise((resolve, reject) => {
+
+            // Fetching the list of available broadcasts first.
+            this._fetchGoogleEndpoint(googleAuth, API_URL_LIVE_BROADCASTS)
+            .then(broadcasts => {
+                // Then fetching all the available live streams that the
+                // user has access to with the broadcasts we retreived
+                // earlier.
+                this._getLiveStreamsForBroadcasts(
+                    googleAuth, broadcasts).then(resolve, reject);
+            }, reject);
+        });
+    }
+
+    /**
+     * Helper method to fetch a Google API endpoint in a generic way.
+     *
+     * @private
+     * @param {Object} googleAuth - The response from a successful Google Auth.
+     * @param {string} endpoint - The endpoint to fetch, including the URL
+     * params if needed.
+     * @returns {Promise}
+     */
+    _fetchGoogleEndpoint(googleAuth, endpoint): Promise<*> {
+        return new Promise((resolve, reject) => {
+            const headers = {
+                Authorization: `Bearer ${googleAuth.accessToken}`
+            };
+
+            fetch(endpoint, {
+                headers
+            }).then(response => response.json())
+            .then(responseJSON => {
+                if (responseJSON.error) {
+                    reject(responseJSON.error.message);
+                } else {
+                    resolve(responseJSON.items || []);
+                }
+            }, reject);
+        });
+    }
+
+    /**
+     * Retrieves the available YouTube streams that are available for the
+     * provided broadcast IDs.
+     *
+     * @private
+     * @param {Object} googleAuth - The response from a successful Google Auth.
+     * @param {Array<Object>} broadcasts - The list of broadcasts that we want
+     * to retreive streams for.
+     * @returns {Promise}
+     */
+    _getLiveStreamsForBroadcasts(googleAuth, broadcasts): Promise<*> {
```

Comment:
I wonder if it's better to only pass what's needed for now, which is the access token, instead of the object. It probably won't become a problem either way.

Reply:
Yep, maybe that's clearer, thanks!

Reply:
Oh, you mean in this method? I though you meant the underlying methods. Do you want me to change this too?

Reply:
I had meant this method (and maybe I was implying other methods too) but only if you see value in doing so.

Reply:
Good point, thanks, I changed it